### ISSUE TITLE
[REVIEW]: GENRE (GPU Elastic-Net REgression): A CUDA-Accelerated Package for Massively Parallel Linear Regression with Elastic-Net Regularization

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2644]
**Submitting author:** @Christopher-Khan (<a href="http://orcid.org/0000-0003-3201-3423">Christopher Khan</a>)
**Repository:** <a href="https://github.com/VU-BEAM-Lab/GENRE" target ="_blank">https://github.com/VU-BEAM-Lab/GENRE</a>
**Version:** v1.0.1
**Editor:** @sjpfenninger
**Reviewer:** @marouenbg, @krystophny
**Archive:** <a href="https://doi.org/10.5281/zenodo.4076520" target="_blank">10.5281/zenodo.4076520</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f/status.svg)](https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f"><img src="https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f/status.svg)](https://joss.theoj.org/papers/ba7eaa02926a24344a680cbd0c5ac87f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/VU-BEAM-Lab/GENRE" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@marouenbg & @krystophny, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sjpfenninger know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/VU-BEAM-Lab/GENRE">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Christopher-Khan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/VU-BEAM-Lab/GENRE">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@Christopher-Khan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @marouenbg, @krystophny it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.1467-9868.2005.00503.x is OK
- 10.18637/jss.v033.i01 is OK
- 10.1109/tuffc.2014.2928 is OK
- 10.1109/tuffc.2015.007004 is OK
- 10.1109/tuffc.2017.2729944 is OK
- 10.1186/1753-6561-6-s2-s10 is OK
- 10.3389/fgene.2013.00270 is OK
- 10.1016/j.neuroimage.2008.08.020 is OK
- 10.1109/isbi.2014.6868131 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf)

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf)

**Comment by marouenbg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by marouenbg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by marouenbg** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by marouenbg** [comment]
Hi @sjpfenninger ,
I finished my review, @Christopher-Khan addressed all of the comments either through additional experiments and implementation, new manuscript sections, and through point-by-point rebuttal.

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
Hi @marouenbg. I just added one more result to the manuscript. The glmnet software package has two algorithm options to choose from, so I also included the benchmark result for the other option as well. It was slower than the option whose benchmark result I had already reported though. In addition, thank you again for doing this review! Your comments were very helpful in improving this submission!

**Comment by sjpfenninger** [comment]
Thanks @marouenbg for the extensive review and @Christopher-Khan for addressing everything thoroughly! krystophny's review is slightly delayed, but we expect by end of month, so stay tuned.

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by krystophny** [comment]
@sjpfenninger I have finished reviewing the paper and testing the software and support the publication in JOSS after two minor changes in the README https://github.com/VU-BEAM-Lab/GENRE/issues/3 . As mentioned the package is relatively small (1000-2000 lines of code) and not very well known yet, but it looks very useful with its unique capability to solve multiple GLMs with elastic net regularization in parallel on GPUs and has the potential as a standard code for that task. The description is done well, and comparable software (without the special parallel feature) is referenced.

**Comment by Christopher-Khan** [comment]
Hi @krystophny. Thank you for your review! In regards to the minor points, my responses are below.

For adding the folders to the MATLAB path, in the bullet point under Windows and Linux installation in the README, I have the following statement, "Assuming the repository is on your system, go to the MATLAB directory that contains the repository folders and add them to your MATLAB path." 

In regards to not needing to specify the cuda path for Linux, I have the following statement in the second bullet point under Linux compilation in the README, "Note that mexcuda might find the CUDA toolkit library even if you do not explicitly type out its path. In addition, note that there might be differences in your path compared to the one shown above, such as in regards to the version of the CUDA toolkit that is being used."

Did you want me to still modify those statements?

**Comment by krystophny** [comment]
@Christopher-Khan in that case it’s fine, now I understand that these things are correct already now. I was too hasty when reading. So all issues resolved!

**Comment by Christopher-Khan** [comment]
@krystophny Great! Thank you!

**Comment by sjpfenninger** [comment]
@marouenbg @krystophny Thanks both for your reviews. Can you both confirm that you recommend to accept the submission? @krystophny can you close VU-BEAM-Lab/GENRE#3 if it is resolved?

**Comment by krystophny** [comment]
@sjpfenninger everything resolved, I confirm the recommendation to accept the submission!

**Comment by marouenbg** [comment]
Hi @sjpfenninger, I have no further comments, I recommend the publication of the software.

**Comment by sjpfenninger** [comment]
@Christopher-Khan I have made some minor edits to the paper in VU-BEAM-Lab/GENRE#4 -- can you check and merge (ideally squash into single commit)? Also - could you adjust the table captions such that they differentiate what we see in the two tables rather than just saying "Benchmark times"?

**Comment by Christopher-Khan** [comment]
Hi @sjpfenninger. I have squashed the edits into a single commit and merged them. I have also changed the title in the second table to differentiate it from the first table's title.

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@Christopher-Khan Great, we're ready to accept this. Can you make a tagged release on GitHub and archive that release, for example on Zenodo or figshare, then report the version number and archive DOI back here in the review issue?

**Comment by Christopher-Khan** [comment]
Hi @sjpfenninger. Great! Here is the Zenodo information: 10.5281/zenodo.4050385. Before archiving, I also made 3 minor edits to the paper unrelated to your edits (just reworded 3 sentences). Thank you!

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@Christopher-Khan great thanks. Can you edit the Zenodo item such that title and author list are the same as for the JOSS paper?

**Comment by Christopher-Khan** [comment]
Hi @sjpfenninger. Sorry for the late response. I made one more edit to the paper. I forgot for the benchmarks to account for the fact that MATLAB code can sometimes run slower the first time it is called than subsequent calls due to MATLAB's execution engine. Therefore, I reran and updated the benchmarks in order to account for this. The times didn't change much, but I still wanted to update them in the paper. I made a new release on Zenodo and detailed the change on there. I have also updated the metadata. Here is the new Zenodo DOI.  DOI: 10.5281/zenodo.4057940 

Thank you!

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
Hi @sjpfenninger. I was just following up. Was there anything else on my end that you needed me to do regarding the submission or the Zenodo archive? Thank you!

**Comment by sjpfenninger** [comment]
@whedon set doi as 10.5281/zenodo.4057940

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by sjpfenninger** [comment]
@whedon set 10.5281/zenodo.4057940 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4057940" target="_blank">10.5281/zenodo.4057940</a> is the archive.

**Comment by sjpfenninger** [comment]
@whedon set v1.0.1 as version

**Comment by whedon** [comment]
OK. v1.0.1 is the version.

**Comment by sjpfenninger** [comment]
@Christopher-Khan Apologies for the delay. This looks good now!

**Comment by sjpfenninger** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.1467-9868.2005.00503.x is OK
- 10.18637/jss.v033.i01 is OK
- 10.1109/tuffc.2014.2928 is OK
- 10.1109/tuffc.2015.007004 is OK
- 10.1109/tuffc.2017.2729944 is OK
- 10.1186/1753-6561-6-s2-s10 is OK
- 10.3389/fgene.2013.00270 is OK
- 10.1016/j.neuroimage.2008.08.020 is OK
- 10.1109/isbi.2014.6868131 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1790

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1790, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Hi @Christopher-Khan! I see that you archive and version are set and up to date. I just looked at your paper and unfortunately it is far too long. A [JOSS paper](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) is to be between 250 and 1000 words. Can you move some of the content from the paper to your documentation?

**Comment by Christopher-Khan** [comment]
Hi @kthyng! Sure. I can move the section titled "Example Benchmark Comparing GENRE with glmnet" to my documentation on GitHub. That will make the paper two pages shorter. Do you want me to make a new Zenodo archive then and give the DOI for that one? I ask because I don't think Zenodo allows for files in archives to be updated if the archive is more than a week old. Thank you!

**Comment by krystophny** [comment]
@Christopher-Khan if you make a new release on github, Zenodo will create a new version of the same entry with a new doi. Probably that’s better than creating a new archive

**Comment by Christopher-Khan** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02644/joss.02644/10.21105.joss.02644.pdf) :page_facing_up: :point_left:

**Comment by Christopher-Khan** [comment]
@krystophny thank you. That's what I meant actually instead of creating a new archive. @kthyng I have moved the section in the JOSS paper titled "Example Benchmark Comparing GENRE with glmnet" from the paper to the section in the README.md file titled "Comparing with Other Packages". The new DOI is below. I hope the paper is better now in terms of its length.

DOI: 10.5281/zenodo.4076520 

Thank you for your time.

**Comment by kthyng** [comment]
@Christopher-Khan Yes looks good! We can proceed now.

**Comment by kthyng** [comment]
@whedon set 10.5281/zenodo.4076520 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4076520" target="_blank">10.5281/zenodo.4076520</a> is the archive.

**Comment by kthyng** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1111/j.1467-9868.2005.00503.x is OK
- 10.18637/jss.v033.i01 is OK
- 10.1109/tuffc.2014.2928 is OK
- 10.1109/tuffc.2015.007004 is OK
- 10.1109/tuffc.2017.2729944 is OK
- 10.1186/1753-6561-6-s2-s10 is OK
- 10.3389/fgene.2013.00270 is OK
- 10.1016/j.neuroimage.2008.08.020 is OK
- 10.1109/isbi.2014.6868131 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1797

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1797, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1314978698991489026) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1798
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02644](https://doi.org/10.21105/joss.02644)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on the new publication @Christopher-Khan! Thanks to editor @sjpfenninger and reviewers @marouenbg and @krystophny for your time and expertise!!

(will close this issue once doi resolves)

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02644/status.svg)](https://doi.org/10.21105/joss.02644)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02644">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02644/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02644/status.svg
   :target: https://doi.org/10.21105/joss.02644
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02644/status.svg)](https://doi.org/10.21105/joss.02644)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by Christopher-Khan** [comment]
Awesome! Thank you @kthyng! Also, thank you @sjpfenninger, @marouenbg, and @krystophny! This review process has been great!

**Comment by Christopher-Khan** [comment]
@whedon set v1.0.2 as version

**Comment by whedon** [comment]
I'm sorry @Christopher-Khan, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by Christopher-Khan** [comment]
@sjpfenninger. Last thing. Do I need to set the version at the top of this review from v1.0.1 to v1.0.2 because that's the newest version that actually got accepted? If so, I believe only editors can do that. Thank you!

**Comment by arfon** [comment]
@whedon set v1.0.2 as version

**Comment by whedon** [comment]
OK. v1.0.2 is the version.

**Comment by Christopher-Khan** [comment]
@arfon Great! Thank you!
